src/grpc-ilp-network-client.web.ts

```diff
@@ -0,0 +1,68 @@
+import { IlpNetworkClient } from './ilp-network-client'
+import { GetBalanceRequest } from './generated/web/ilp/get_balance_request_pb'
+import { GetBalanceResponse } from './generated/web/ilp/get_balance_response_pb'
+import { SendPaymentRequest } from './generated/web/ilp/send_payment_request_pb'
+import { SendPaymentResponse } from './generated/web/ilp/send_payment_response_pb'
+import isNode from './utils'
+import { BalanceServiceClient } from './generated/web/ilp/balance_service_grpc_web_pb'
+import { IlpOverHttpServiceClient } from './generated/web/ilp/ilp_over_http_service_grpc_web_pb'
+
+class GrpcIlpNetworkClientWeb implements IlpNetworkClient {
+  private readonly balanceClient: BalanceServiceClient
+
+  private readonly paymentClient: IlpOverHttpServiceClient
+
+  public constructor(grpcURL: string) {
+    if (isNode()) {
+      try {
+        // This polyfill hack enables XMLHttpRequest on the global node.js state
+        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+        // @ts-ignore eslint-disable-line
+        global.XMLHttpRequest = require('xhr2') // eslint-disable-line
+      } catch {
+        // Swallow the error here for browsers
+      }
+    }
+    // FIXME wrong credentials
+    this.balanceClient = new BalanceServiceClient(grpcURL, null, null)
+    this.paymentClient = new IlpOverHttpServiceClient(grpcURL, null, null)
+  }
+
+  getBalance(request: GetBalanceRequest): Promise<GetBalanceResponse> {
+    return new Promise((resolve, reject): void => {
+      // FIXME should metadata be undefined?
+      this.balanceClient.getBalance(request, undefined, (error, response) => {
```

Comment:
The web compiler just doesn't generate the non-verbose version of this method. If you look at the implementation, omitting this param just passes null.  If it bugs you that there's an extra param here you can also change the non-web code.

Metadata is just an arbitrary set of key value pairs. You can create one by:
```
import grpc from 'grpc'

const m = new grpc.Metadata()
```

Reply:
I wouldn't recommend importing `grpc` in the .web files for compatibility reasons. I believe we just pass `{}` in the legacy / rippled proto clients

Reply:
ah, sorry about that. 

If you need this metadata let us know and we can help you figure out how to get it set up.

Reply:
It's probably a TBD since there are some aspects about how we'll do auth that are up in the air right now.